### ISSUE TITLE
Not found error when trying to run model

### ISSUE THREAD

**Issue by jda91** [tensorflow/serving#684]
**Trying bazel build -c opt tensorflow_serving/... doesn't work so, it always gives me a SIMD not enabled (maybe define GEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK in the source code when x86_64) error, so I had to use bazel build -c opt --copt=-msse4.1 --copt=-msse4.2 tensorflow_serving/... and that built wihtout any errors. But now I always get the same error when trying to test the running model, in fact I get the error trying to run anything in the container. I had to export the model myself by adding savedmodelbuilder code into the retrain script for inception. I'm on an Ubunutu PC, but I tried running tensorflow serving on a mac pro and got the same error when trying bazel build -c opt tensorflow_serving/... so again I had to use bazel build -c opt --copt=-msse4.1 --copt=-msse4.2 tensorflow_serving/..., and again the same exact error listed below**

```
root@42fd2c27af9d:/serving# bazel-bin/tensorflow_serving/example/inception_client --server=localhost:9000 --image=./Xiang_Xiang_panda.jpg
Traceback (most recent call last):
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/tf_serving/tensorflow_serving/example/inception_client.py", line 56, in <module>
    tf.app.run()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 129, in run
    _sys.exit(main(argv))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/tf_serving/tensorflow_serving/example/inception_client.py", line 50, in main
    tf.contrib.util.make_tensor_proto(data, shape=[1]))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 53, in __getattr__
    module = self._load()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 42, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/__init__.py", line 81, in <module>
    from tensorflow.contrib.eager.python import tfe as eager
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/eager/python/tfe.py", line 75, in <module>
    from tensorflow.contrib.eager.python.datasets import Iterator
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/eager/python/datasets.py", line 23, in <module>
    from tensorflow.contrib.data.python.ops import prefetching_ops
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/prefetching_ops.py", line 25, in <module>
    resource_loader.get_path_to_datafile("../../_prefetching_ops.so"))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/util/loader.py", line 55, in load_op_library
    ret = load_library.load_op_library(path)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename, status)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal26fixed_address_empty_stringB5cxx11E
```

**When I check the running log I can see that my model is running:**
```

2017-12-09 01:24:00.397485: I tensorflow_serving/model_servers/main.cc:147] Building single TensorFlow model file config:  model_name: inception model_base_path: /tmp/new5
2017-12-09 01:24:00.397670: I tensorflow_serving/model_servers/server_core.cc:439] Adding/updating models.
2017-12-09 01:24:00.397696: I tensorflow_serving/model_servers/server_core.cc:490]  (Re-)adding model: inception
2017-12-09 01:24:00.498119: I tensorflow_serving/core/basic_manager.cc:705] Successfully reserved resources to load servable {name: inception version: 1}
2017-12-09 01:24:00.498154: I tensorflow_serving/core/loader_harness.cc:66] Approving load for servable version {name: inception version: 1}
2017-12-09 01:24:00.498169: I tensorflow_serving/core/loader_harness.cc:74] Loading servable version {name: inception version: 1}
2017-12-09 01:24:00.498189: I external/org_tensorflow/tensorflow/contrib/session_bundle/bundle_shim.cc:360] Attempting to load native SavedModelBundle in bundle-shim from: /tmp/new5/1
2017-12-09 01:24:00.498203: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:236] Loading SavedModel from: /tmp/new5/1
2017-12-09 01:24:00.623487: I external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2 FMA
2017-12-09 01:24:00.743901: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:155] Restoring SavedModel bundle.
2017-12-09 01:24:00.798587: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:190] Running LegacyInitOp on SavedModel bundle.
2017-12-09 01:24:00.805405: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:284] Loading SavedModel: success. Took 307196 microseconds.
2017-12-09 01:24:00.805517: I tensorflow_serving/core/loader_harness.cc:86] Successfully loaded servable version {name: inception version: 1}
2017-12-09 01:24:00.810840: I tensorflow_serving/model_servers/main.cc:288] Running ModelServer at 0.0.0.0:9000 ...
```

**Comment by 12wzy** [comment]
+1. I get the same error  when I try to run mnist_saved_model.

**Comment by jda91** [comment]
Can someone please help us out, I have tried it on a windows PC, Mac Pro, and PC running Ubuntu, all give the same error.

**Comment by chrisolston** [comment]
Just a quick update to let folks know that we're aware of this issue and looking into it. I think it's something to do with the TF layer or the synchronization between TF-Serving and TF; we are investigating.

**Comment by eugene070** [comment]
Ubuntu 16
built:
`bazel build -c opt -msse4 tensorflow/examples/image_retraining:retrain `

then tried to run with:
`bazel-bin/tensorflow/examples/image_retraining/retrain --image_dir ~/my_images`

and got:
`Traceback (most recent call last):
  File "/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/examples/image_retraining/retrain.py", line 117, in <module>
    from tensorflow.contrib.quantize.python import quant_ops
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/__init__.py", line 82, in <module>
    from tensorflow.contrib.eager.python import tfe as eager
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/eager/python/tfe.py", line 76, in <module>
    from tensorflow.contrib.eager.python.datasets import Iterator
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/eager/python/datasets.py", line 23, in <module>
    from tensorflow.contrib.data.python.ops import prefetching_ops
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/prefetching_ops.py", line 25, in <module>
    resource_loader.get_path_to_datafile("../../_prefetching_ops.so"))
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/util/loader.py", line 55, in load_op_library
    ret = load_library.load_op_library(path)
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename, status)
  File "/tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /tensorflow/bazel-bin/tensorflow/examples/image_retraining/retrain.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal26fixed_address_empty_stringB5cxx11E
`

Any ideas to solve this issue?

**Comment by rish321** [comment]
+1. Any solution for this?

**Comment by PenalizedEM** [comment]
+1. Any solution

**Comment by jda91** [comment]
Ive resorted to running my model on flask, though I would have loved to use tensorflow serving, I just can't get it to work

**Comment by soar-zhengjian** [comment]
I get the same error when I try to run mnist_saved_model by bazel. But It's ok when I run it by python.

**Comment by Cristiano-3** [comment]
I get the same error when I try to run bazel-bin/tensorflow_serving/example/inception_saved_model /tmp/inception_model.

**Comment by vmitt** [comment]
Hello All, I am also getting the same error. Also, can somebody confirm, if this issue is now coming because of the protobuf_archive changes that are done by Google recently. Earlier(a couple of months back), this thing used to work but recently this issue is now coming. ----------- 

root@902ea6be6d06:/serving# bazel-bin/tensorflow_serving/example/inception_saved_model --checkpoint_dir=tensorflow/incep
tion-v3 --export_dir=inception-export
Traceback (most recent call last):
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/tf_serving/tensorflow_serving/example/inception_saved_model.py", line 202, in <module>
    tf.app.run()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 124, in run
    _sys.exit(main(argv))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/tf_serving/tensorflow_serving/example/inception_saved_model.py", line 198, in main
    export()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/tf_serving/tensorflow_serving/example/inception_saved_model.py", line 91, in export
    table = tf.contrib.lookup.index_to_string_table_from_tensor(class_tensor)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 53, in __getattr__
    module = self._load()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 42, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/__init__.py", line 82, in <module>
    from tensorflow.contrib.eager.python import tfe as eager
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/eager/python/tfe.py", line 75, in <module>
    from tensorflow.contrib.eager.python.datasets import Iterator
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/eager/python/datasets.py", line 23, in <module>
    from tensorflow.contrib.data.python.ops import prefetching_ops
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/prefetching_ops.py", line 25, in <module>
    resource_loader.get_path_to_datafile("../../_prefetching_ops.so"))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/util/loader.py", line 55, in load_op_library
    ret = load_library.load_op_library(path)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename, status)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /serving/bazel-bin/tensorflow_serving/example/inception_saved_model.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal26fixed_address_empty_stringB5cxx11E

**Comment by fengdragon** [comment]
@soar-zhengjian Got the same error...while using saved_model, could you help to introduce how you made it work via python? Thanks a lot!

**Comment by fengdragon** [comment]
@chrisolston is there any update?

**Comment by chrisolston** [comment]
Tensorflow Core still has its last green build at Dec 7 so we aren't able to sync the submodule to a newer version at this time. Once that issue gets resolved we will sync submodules and hopefully get things unstuck here. Thanks, all, for your patience.

**Comment by vmitt** [comment]
Thanks Chrisolston. Do we have any tentative timelines for it, so that I could ask my customer to wait till that long for production deployment using TF Serving. Thanks so much for the update.

@fengdragon, I can help you with the python version of this inception code running and retraining it with our labels. You can just send me an email and its quite easy to do. Please note that this does not use any Tensorflow Serving or Bazel build concepts, but a general old timer code.

**Comment by chrisolston** [comment]
It's hard to say with these sorts of things, but now that the holiday season has passed it will hopefully get resolved in the next week or so.

**Comment by chrisolston** [comment]
You can also consider using our 1.4 release rather than head, if that works for you.

**Comment by fengdragon** [comment]
@vmitt Dear vmitt, I had sent an email to your gmail, Thanks for your help and time!

**Comment by exp0nge** [comment]
+1

**Comment by ash-lat** [comment]
+1 -rasbian strech desktop 
tf 1.4

**Comment by hery** [comment]
+1 using the Dockerfile.devel container

**Comment by xushenkun** [comment]
+1 ubuntu 16.04.2 python 2.7.12 tf 1.4.1

**Comment by offbye** [comment]
same problems

**Comment by TonyChouZJU** [comment]
Build label: 0.9.0
TF: 1.5.0 CPU
Python 2.7.14 :: Anaconda custom (64-bit)
Ubuntu 16.04 TLS

The **same error** arise when I try to run inception-client using **both bazel and tensorflow-serving-api**.

bazel built
```
bazel-bin/tensorflow_serving/example/inception_client --server=localhost:9000 --image=/path-to-image

python tensorflow_serving/example/inception_client.py --server=localhost:9000 --image=/path-to-image
```
tensorflow-serving-apis
```
tensorflow_model_server --server=localhost:9000 --image=/path-to-image

python tensorflow_serving/example/inception_client.py --server=localhost:9000 --image=/path-to-image
```

The error is the same as above mentioned
>  File "/home/zyb/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /home/zyb/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal9ArenaImpl10AddCleanupEPvPFvS3_E

----

But it **workes well** with bazel server and python client
```
bazel-bin/tensorflow_serving/example/inception_client --server=localhost:9000 --image=/path-to-image

python tensorflow_serving/example/inception_client.py --server=localhost:9000 --image=/path-to-image
```
Any solutions? Thanks in advance.

**Comment by recolic** [comment]
reproduced at ubuntu 16.04 in docker, following [this guide](https://www.tensorflow.org/serving/serving_inception)
```
root@36f3b01e432f:/serving# bazel-bin/tensorflow_serving/example/inception_client --server=localhost:9000 --image=./cat-img1.jpg
<Waiting for a minute...>
Traceback (most recent call last):
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/tf_serving/tensorflow_serving/example/inception_client.py", line 56, in <module>
    tf.app.run()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 124, in run
    _sys.exit(main(argv))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/tf_serving/tensorflow_serving/example/inception_client.py", line 50, in main
    tf.contrib.util.make_tensor_proto(data, shape=[1]))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 53, in __getattr__
    module = self._load()
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 42, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/__init__.py", line 83, in <module>
    from tensorflow.contrib.eager.python import tfe as eager
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/eager/python/tfe.py", line 78, in <module>
    from tensorflow.contrib.eager.python.datasets import Iterator
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/eager/python/datasets.py", line 23, in <module>
    from tensorflow.contrib.data.python.ops import prefetching_ops
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/prefetching_ops.py", line 25, in <module>
    resource_loader.get_path_to_datafile("../../_prefetching_ops.so"))
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/util/loader.py", line 55, in load_op_library
    ret = load_library.load_op_library(path)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename, status)
  File "/serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /serving/bazel-bin/tensorflow_serving/example/inception_client.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal9ArenaImpl10AddCleanupEPvPFvS3_E
```

**Comment by HelloSeeing** [comment]
+1
ubuntu 14.04
python 2.7.6
tf 1.4.1

**Comment by TharinduDR** [comment]
Rather than cloning from master branch clone from 'r1.4' it works then

**Comment by hiepph** [comment]
Solution, anyone?
Arch Linux
python 3.6.4 (Anaconda)
Tensorflow GPU 1.4.1 (installed from Anaconda cloud)

**Comment by VeryVeryGoodNews** [comment]
Hello @TharinduDR , can you post your code of how you branch cloned from r1.4? I looked at prior threads on how to clone a prior version in general, but those threads were rife with confusing and conflicting answers. Thanks

**Comment by galv** [comment]
I am also seeing the error described by @jda91 

No resolution to this after over two months?

Maybe someone should git bisect to find the exact commit causing the problem. Too bad compiling takes so long.

**Comment by TharinduDR** [comment]
@VeryVeryGoodNews 
git clone -b <branch> <remote_repo>

**Comment by charmoniumQ** [comment]
@VeryVeryGoodNews I could not get r1.4 to compile at all.

`
$ git checkout r1.4 && git submodule update --recursive
$ bazel build
Loading: 
Loading: 0 packages loaded
ERROR: error loading package '': Encountered error while reading extension file 'tensorflow/workspace.bzl': no such package '@org_tensorflow//tensorflow': No WORKSPACE file found in /home/admin/.cache/bazel/_bazel_admin/63fe3b6347eadc01a6b5a65081712706/external/org_tensorflow
ERROR: error loading package '': Encountered error while reading extension file 'tensorflow/workspace.bzl': no such package '@org_tensorflow//tensorflow': No WORKSPACE file found in /home/admin/.cache/bazel/_bazel_admin/63fe3b6347eadc01a6b5a65081712706/external/org_tensorflow
INFO: Elapsed time: 0.017s
FAILED: Build did NOT complete successfully (0 packages loaded)
`
I also tried checking out the `WORKSPACE` from `master`, although that might defeat the purpose. Even that did not work.

**Comment by annalehnchen** [comment]
does someone has an other solution than switching back to the 1.4-branch?

**Comment by gareth-ferneyhough** [comment]
@charmoniumQ I got r1.4 working. I believe this is what worked for me:
```
git clone -b r1.4 --recurse-submodules https://github.com/tensorflow/serving
cd serving/tensorflow
./configure
cd ..
bazel build
```

**Comment by kashifmin** [comment]
@gareth-ferneyhough 
Thanks. But I do not want to rebuild the whole thing again. Is there anyway to get this working on master branch source?

**Comment by cyliustack** [comment]
I just solved the problem by `git checkout v1.5.0-rc1` and then rebuilding it.

**Comment by anpark** [comment]
any solution? same error:undefined symbol: _ZN6google8protobuf8internal9ArenaImpl15AllocateAlignedEm

**Comment by zhiz21** [comment]
Same error:
**OS:** Amazon Linux
**TF 1.7** release
**Python:** 2.7
Compiled from source with CUDA support on AWS P3.16xlarge instance.
```
undefined symbol: _ZN6google8protobuf8internal9ArenaImpl15AllocateAlignedEm
```

**Comment by kexu935** [comment]
I came up with the same issue. Any update? @chrisolston  Thanks.

**Comment by likan999** [comment]
Hitting the same error when following profiling instruction and has symbol undefined error when running the line "with tf.contrib.tfprof.ProfileContext('/tmp/train_dir') as pctx:"

**Comment by SatioO** [comment]
Training model...
Extracting /tmp/train-images-idx3-ubyte.gz
Extracting /tmp/train-labels-idx1-ubyte.gz
Extracting /tmp/t10k-images-idx3-ubyte.gz
Extracting /tmp/t10k-labels-idx1-ubyte.gz
2018-04-03 10:16:59.838219: I external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
Traceback (most recent call last):
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/tf_serving/tensorflow_serving/example/mnist_saved_model.py", line 147, in <module>
    tf.app.run()
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/tf_serving/tensorflow_serving/example/mnist_saved_model.py", line 74, in main
    table = tf.contrib.lookup.index_to_string_table_from_tensor(
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 53, in __getattr__
    module = self._load()
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/python/util/lazy_loader.py", line 42, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/__init__.py", line 84, in <module>
    from tensorflow.contrib.eager.python import tfe as eager
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/eager/python/tfe.py", line 78, in <module>
    from tensorflow.contrib.eager.python.datasets import Iterator
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/eager/python/datasets.py", line 23, in <module>
    from tensorflow.contrib.data.python.ops import prefetching_ops
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/prefetching_ops.py", line 25, in <module>
    resource_loader.get_path_to_datafile("../../_prefetching_ops.so"))
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/util/loader.py", line 55, in load_op_library
    ret = load_library.load_op_library(path)
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/python/framework/load_library.py", line 58, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename, status)
  File "/serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/python/framework/errors_impl.py", line 516, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: /serving/bazel-bin/tensorflow_serving/example/mnist_saved_model.runfiles/org_tensorflow/tensorflow/contrib/data/python/ops/../../_prefetching_ops.so: undefined symbol: _ZN6google8protobuf8internal9ArenaImpl10AddCleanupEPvPFvS3_E


Getting same issue with Mnist model.

**Comment by lyhcloud** [comment]
+1 on raspbian stretch
both tf 1.5 and 1.7

**Comment by cpmgs** [comment]
+1
python 3.5.2
tf 1.7
Ubuntu 16.04

**Comment by hanm** [comment]
I think the issue is caused by commit https://github.com/tensorflow/tensorflow/commit/cd81bc8e09c7f551911276c5bfaafa6930f1961f, which is later fixed by https://github.com/tensorflow/tensorflow/commit/36ec749ec79c2313924666a1c5324620e493d0c4. This fix is not available in the Tensorflow that Tensorflow Serving master branch is referencing. 

One workaround is to manually update the git commit hash in [Tensorflow Serving Workspace file](https://github.com/tensorflow/serving/blob/master/WORKSPACE#L15) to reference a version of Tensorflow that has this issue fixed. That works for me locally at least (note you can just disable the sha256 if you don't want to generate one), and I can export the Inception model now. As a reference the commit has I am using is the head of r1.8 as of today (3970b47da568a783818f43ec9f8df5acf692fe7c).

**Comment by sflinter** [comment]
Thanks @hanm - I've been pulling my hair out over this. This works for me. I've updated the WORKSPACE file as follows:

```
tensorflow_http_archive(
    name = "org_tensorflow",
    sha256 = "ac50a03c883cffbac01709e553736273ec46011f8df66efeb231eafeed4cc147",
    git_commit = "3970b47da568a783818f43ec9f8df5acf692fe7c",
)

```
This references the v1.8.0-rc0 tag of Tensorflow (obviously the same one you're using).

With this change, the rest of the instructions for building and deploying the inception model work for me.

Interestingly, I found that all of the tests failed, but that that didn't seem to stop anything else from working. Any ideas why that might be?

**Comment by hanm** [comment]
@sflinter I forgot to mention another place I have to patch Tensorflow Serving to make the model server build (with the v1.8 commit hash of Tensorflow) because commit https://github.com/tensorflow/tensorflow/commit/46231cf242c19d74af75370eefd9e9b7c504c08a#diff-94e5d6085b8a65623e69b69ed90c1767 in Tensorflow moved the port namespace definition to a different file and there are a couple of places in TensorflowServing where port was referenced need to be updated accordingly. 

Not sure if this relates to what you saw about the test failures. When I have a chance I'll try run the tests and see what I get. For now I am happy as at least the tutorial works for me...

**Comment by jupiterianian** [comment]
@hanm at least two last commits (as of now :)) are referring to tensorflow's commits with needed corrections. Just checkout it and further no need to play around with hashes in the WORKSPACE (and further with port definitions introduced you've mentioned). 

Almost the whole serving's tree can be compiled with bazel 0.11.x and python 3.6.x. The only issue I saw is tar and deb packaging of model server, therefore it should be removed from BUILD (e.g. `sed -i '/pkg:pkg.bzl/Q' tensorflow_serving/model_servers/BUILD` either in a different manner) to compile the whole tree.

**Comment by hanm** [comment]
@jupiterianian Yes, the inception model serving tutorial now works with top of the trunk today with commit (https://github.com/tensorflow/serving/commit/c99e6082db5bb8ac20d7653a36fdca7396a3ea73). Glad that recent TF changes are merged that fix the issue!

**Comment by mit2nil** [comment]
+1
python 3.5.2
tf 1.7
Ubuntu 16.04
cuda 9.1 
cudnn 7.1.3

**Comment by ibzip** [comment]
Seeing the same issue on 21st may with r1.6 while trying to run it inside the docker container built with the image provided in the tensorflow-serving tutorial with Kubernetes. Is there any official workaround/solution for this other than using r1.4 or manually editing the workspace file?

**Comment by gautamvasudevan** [comment]
This should be resolved with the release of 1.8. An example of a functional build is the Dockerfile.devel build (referenced [here](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/g3doc/docker.md))

**Comment by CLIsVeryOK** [comment]
someone get it work at [https://github.com/tensorflow/serving/issues/421], but i get the same error after comment __init___.py as they suggest.
